### PR TITLE
MDS-488 feature/updated create/edit mine

### PR DESCRIPTION
**Pull request by mmumby**

### CODE REVIEW

File: frontend/src/actionCreators/mineActionCreator.js [hunk 1]
```diff
@@ -107,3 +107,19 @@ export const fetchStatusOptions = () => (dispatch) => {
       dispatch(hideLoading('modal'));
     });
 };
+
+export const fetchRegionOptions = () => (dispatch) => {
+  dispatch(request(reducerTypes.GET_REGION_OPTIONS));
+  dispatch(showLoading('modal'));
+  return axios.get(ENVIRONMENT.apiUrl + API.MINE_REGION, createRequestHeader())
```
Comment (NWCalvank) [inline frontend/src/actionCreators/mineActionCreator.js]: Thoughts on string interpolation in this codebase? I don't think i've seen any instances of it being used.

This line is likely nicer visually this way, but I'm curious about the concatenation vs interpolation pattern across the codebase.

File: frontend/src/actionCreators/mineActionCreator.js [hunk 2]
```diff
@@ -107,3 +107,19 @@ export const fetchStatusOptions = () => (dispatch) => {
...
+    .then((response) => {
+      dispatch(success(reducerTypes.GET_REGION_OPTIONS));
+      dispatch(mineActions.storeRegionOptions(response.data));
+      dispatch(hideLoading('modal'));
+    })
+    .catch((err) => {
```
Comment (NWCalvank) [inline frontend/src/actionCreators/mineActionCreator.js]: Thoughts on destructuring out `response` here?

File: frontend/src/components/Forms/UpdatePermitteeForm.js [hunk 3]
```diff
@@ -30,7 +28,7 @@ const permitteeOptions = (permit) => {
   permit.map((obj) => {
     const data = {
       value: obj.permittee[0].permittee_guid.concat(', ', obj.permit_guid),
-      option: obj.permittee[0].party.name.concat(", ", obj.permit_no)
+      label: obj.permittee[0].party.name.concat(", ", obj.permit_no)
```
Comment (NWCalvank) [inline frontend/src/components/Forms/UpdatePermitteeForm.js]: Destructure `obj` and set label to `${permittee.party.name},${permit_no}`

Thoughts?

Also, do we need to validate that `.permit_no` or `.name` will be present before accessing them, or are they always required?

File: frontend/src/components/common/RenderCheckbox.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+const RenderCheckbox = ({
+  id,
+  input,
+  label,
+  meta: { touched, error},
+}) => (
+  <Form.Item 
+  validateStatus={(touched ? (error && 'error') : '')}
+>
```
Comment (NWCalvank) [inline frontend/src/components/common/RenderCheckbox.js]: Looks like this should be vertically aligned

File: frontend/src/components/common/RenderCheckbox.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  <Checkbox 
+    id={id} 
+    checked={input.value || false}
```
Comment (NWCalvank) [inline frontend/src/components/common/RenderCheckbox.js]: Is the `|| false` necessary here? My impression would be that `input.value` gets coerced?

But if it doesn't, how about `checked={Boolean(input.value)}` to be a bit more explicit about the issue?

File: frontend/src/components/dashboard/Dashboard.js [hunk 6]
```diff
@@ -253,14 +254,15 @@ export class Dashboard extends Component {
   }
 
   render() {
+    console.log(this.props.mineRegionOptions);
```
Comment (NWCalvank) [inline frontend/src/components/dashboard/Dashboard.js]: Remove console.log

File: frontend/src/components/mine/MineHeader.js [hunk 7]
```diff
@@ -47,16 +48,18 @@ class MineHeader extends Component {
...
   
-  openModal(event, mineStatusOptions, onSubmit, title, mine) {
+  openModal(event, mineStatusOptions, mineRegionOptions, onSubmit, title, mine) {
     event.preventDefault();
     const initialValues = {
       "name": mine.mine_detail[0] ? mine.mine_detail[0].mine_name : null,
       "latitude": mine.mine_location[0] ? mine.mine_location[0].latitude : null,
       "longitude": mine.mine_location[0] ? mine.mine_location[0].longitude : null,
       "mine_status": mine.mine_status[0] ? mine.mine_status[0].status_values : null,
+      "major": mine.mine_detail[0] ? mine.mine_detail[0].major : false,
```
Comment (NWCalvank) [inline frontend/src/components/mine/MineHeader.js]: Looks like we're inconsistent around single vs double-quotes. Do you have a preference?

File: frontend/src/components/mine/MineHeader.js [hunk 8]
```diff
@@ -84,10 +87,12 @@ class MineHeader extends Component {
               isDropdown 
               overlay={menu} 
               string={<Icon type="ellipsis" theme="outlined" style={{fontSize: '30px'}}/>}
-            />
+              />
```
Comment (NWCalvank) [inline frontend/src/components/mine/MineHeader.js]: This looks like an indentation error

File: frontend/src/tests/components/common/RenderCheckbox.spec.js [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+    },
+  };
+}
+
+beforeEach(() => {
+  setupProps();
+});
+
+describe('RenderCheckbox', () => {
+
```
Comment (NWCalvank) [inline frontend/src/tests/components/common/RenderCheckbox.spec.js]: The other spec file doesn't have newline padding around the `it` blocks. Which would you prefer? If we are using the Airbnb eslint config, they disallow newline padding.

https://github.com/airbnb/javascript#whitespace--padded-blocks

File: frontend/src/tests/mocks/dataMocks.js [hunk 10]
```diff
@@ -277,6 +291,29 @@ export const STATUS_OPTIONS = {
   ]
 };
 
+export const REGION_OPTIONS = { 
+  'options': [{
```
Comment (NWCalvank) [inline frontend/src/tests/mocks/dataMocks.js]: Any reason for using single quotes on these keys? They shouldn't be needed, and I don't think we're generally adding them in other files

File: frontend/src/components/common/RenderCheckbox.js (hunk 4 above)
Comment (mmumby) [reply frontend/src/components/common/RenderCheckbox.js]: Good catch, alignment was very off.

File: frontend/src/components/common/RenderCheckbox.js (hunk 5 above)
Comment (mmumby) [reply frontend/src/components/common/RenderCheckbox.js]: false is not needed.

File: frontend/src/components/dashboard/Dashboard.js (hunk 6 above)
Comment (mmumby) [reply frontend/src/components/dashboard/Dashboard.js]: went through and removed them all this am.

File: frontend/src/tests/components/common/RenderCheckbox.spec.js (hunk 9 above)
Comment (mmumby) [reply frontend/src/tests/components/common/RenderCheckbox.spec.js]: no newline padding, updated spec files that have it.

File: frontend/src/tests/mocks/dataMocks.js (hunk 10 above)
Comment (mmumby) [reply frontend/src/tests/mocks/dataMocks.js]: single quotes are removed, these are my mock objects, Usually copy and pasted from somewhere, will pay more attention to consistency!!

Comment (mmumby) [reply frontend/src/tests/mocks/dataMocks.js]: also converted all the values to double quotes

File: frontend/src/components/mine/MineHeader.js (hunk 8 above)
Comment (mmumby) [reply frontend/src/components/mine/MineHeader.js]: fixed.

File: frontend/src/components/mine/MineHeader.js (hunk 7 above)
Comment (mmumby) [reply frontend/src/components/mine/MineHeader.js]: double quotes are preferred throughout.

Single quotes on all imports,

Will work towards removing these inconsistencies.

**Pull request merged**